TFNetworkRecLayer.py

```diff
@@ -3149,24 +3268,58 @@ def transform_config_dict(cls, d, network, get_layer):
     :param TFNetwork.TFNetwork network:
     :param ((str) -> LayerBase) get_layer: function to get or construct another layer
     """
-    if not network.search_flag and not d.get("scheduled_sampling"):
-      # In the dependency graph, we don't want it.
-      # This can enable some optimizations in the RecLayer.
-      # We do it here because we should know about the deps early in the template creation in RecLayer.
-      d["from"] = []
+    if isinstance(d["target"], str):
+      d["target"] = [d["target"]]
+    if not network.search_flag:
+      network.used_data_keys.update(d["target"])
```

Comment:
This is a bit hacky. We need to find out during template creation which targets are used in the sub-net. I'm not sure where it is best to set this. We could also just set mark_data_key_as_used=True in get_out_data_from_opts(). Because there is no place in the code where we call get_out_data_from_opts() and then end up not using the target, I think. Or maybe make mark_data_key_as_used a parameter of get_out_data_from_opts()...